### PR TITLE
Product Gallery Dialog: Refactor Interactivity API actions

### LINKED ISSUE #11311: Product Gallery: Refactor handleClick action
While working on https://github.com/woocommerce/woocommerce-blocks/pull/11087 @roykho pointed out that the current [handleClick](https://github.com/woocommerce/woocommerce-blocks/blob/42c915ed01ea3b530cbbb23373d0c15e1ecc9ae9/assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image/frontend.tsx#L68) action that opens the Dialog could use a different name and should be located in a different place (on the parent, Product Gallery level).

References: https://github.com/woocommerce/woocommerce-blocks/pull/11087#issuecomment-1766300437 and https://github.com/woocommerce/woocommerce-blocks/pull/11087#discussion_r1358519210

### PR DESCRIPTION
**Pull request by danieldudzic**
This PR changes the Product Gallery Dialog Interactivity API action names and location.

## What

Fixes #11311

## Why

Since the `openDialog` action is shared across components it doesn't make sense for it to live inside of the `Large Image`.

The `openDialog` and `closeDialog` names are a bit more informative about the actual actions performed by the action.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. With a block theme, open `Appearance > Editor`.
2. Open `Templates > Single Product`.
3. Select `Convert to Blocks` if you are using the Classic Templates.
4. Add the `Product Gallery` block.
5. Ensure the Thumbnails are enabled.
6. Open a product page on the frontend **with multiple product images** (more than `Number of Thumbnails` setting) and test the frontend. Make sure the last thumbnail displays an overlay that opens the Dialog on click.
7. Make sure there are no regressions and the Dialog can still be triggered by clicking the Large Image.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Add suggested changelog entry here.

### CODE REVIEW

**Review by roykho — CHANGES_REQUESTED** [review]
So there is a regression here. I can no longer click on the gallery thumbnails to change the main image.

File: src/BlockTypes/ProductGalleryThumbnails.php [hunk 1]
```diff
@@ -157,7 +157,7 @@ protected function render( $attributes, $content, $block ) {
 							if ( $processor->next_tag( 'img' ) ) {
 								$processor->set_attribute(
 									'data-wc-on--click',
-									'actions.woocommerce.thumbnails.handleClick'
+									'actions.woocommerce.thumbnails.dialog.openDialog'
```
Comment (thealexandrelara) [inline src/BlockTypes/ProductGalleryThumbnails.php]: I believe that when changing the `handleClick` for the dialog, the editor also changed this that is related to the `thumbnails` by mistake and it causes the Thumbnails to not be clickable anymore.

**Review by thealexandrelara — CHANGES_REQUESTED** [review]
Looks good! I added a comment regarding the `handleClick` function for the Thumbnails block that is currently not working.

**Comment by danieldudzic** [comment]
@roykho @thealexandrelara Thank you for your reviews. I have now fixed the regression and updated other actions too, to follow the same semantic naming pattern. Please let me know if you'd change anything.

**Review by roykho — COMMENTED** [review]
So I guess the only concern I have is the consistency. Because `dialog` has its own sub action but the others do not. So either we use sub actions for example `dialog` for all things housed inside the dialog component and for example `nextPrevious` for all things that belong there such as the `selectNextImage`...etc.

**Comment by thealexandrelara** [comment]
> So I guess the only concern I have is the consistency. Because `dialog` has its own sub action but the others do not. So either we use sub actions for example `dialog` for all things housed inside the dialog component and for example `nextPrevious` for all things that belong there such as the `selectNextImage`...etc.

I agree with you. Since we now have more descriptive names (`openDialog` and `closeDialog`), maybe we don't need the `dialog` key anymore?

**Comment by danieldudzic** [comment]
> I agree with you. Since we now have more descriptive names (`openDialog` and `closeDialog`), maybe we don't need the `dialog` key anymore?

That makes sense and would be my preference, too.

**Comment by danieldudzic** [comment]
I'm closing this PR, as this is being already addressed in the migration to the new Interactivity API store: https://github.com/woocommerce/woocommerce-blocks/pull/11721

**Pull request closed without merging**